python/code.qmd

```diff
@@ -0,0 +1,406 @@
+---
+title: "Writing Python code"
+description: "Guidelines on how to write and style Python code."
+---
+
+What Python code to write is based on the design you've created and
+isn't something that a guide can help with. However, *how* you write
+Python code *is* something that a guide can help with. This page is a
+guide to some common and recommended practices for writing Python
+code, including lessons we've learned.
+
+## General coding style
+
+When writing Python code, you should follow a common and
+well-established style guide. The official style guide for Python is
+called [PEP 8](https://peps.python.org/pep-0008/), and it is a
+comprehensive guide on different aspects of writing Python code. So, you
+should start by reading the PEP 8 guide to learn some basics on the style of
+Python code.
+
+However, it doesn't cover everything. For additional style guidelines,
+there are several other guides you can refer to. The one we
+recommend and use is the [Google Style
+Guide](https://google.github.io/styleguide/pyguide.html). This is the
+second resource you should use to learn about how to write effective and
+well-styled Python code.
+
+Thankfully, there are many aspects of both PEP 8 and the Google Style
+Guide that can be automatically checked and fixed. The tools we use and
+recommend are [Ruff](https://decisions.seedcase-project.org/why-ruff/)
+for general styling and [mypy](https://mypy.readthedocs.io/en/stable/)
+for type checking.
+
+To use Ruff and mypy, you need to be in a Python project. If you have
+created a Python project with our
+[`template-python-package`](https://github.com/seedcase-project/template-python-package),
+then running Ruff can be done through our
+[justfile](https://just.systems/man/en/). In your terminal, you can run
+this:
+
+``` {.bash filename="Terminal"}
+just format-python
+```
+
+Which will run Ruff and format your Python code. Then, to check if your code
+needs manual fixes using Ruff and mypy, you can check the code with:
+
+``` {.bash filename="Terminal"}
+just check-python
+```
+
+::: callout-tip
+If you didn't use our template nor do you want to add Ruff to your
+project's dependencies, you can use Ruff through
+[uv](https://docs.astral.sh/uv/):
+
+``` {.bash filename="Terminal"}
+uvx ruff format
+# Tries to fix the linted issues, a bit different from `format`
+uvx ruff check --fix
+uvx ruff check
+uvx mypy .
+```
+:::
+
+## Where to write code {#where-to-write-code}
```

Comment:
Stumbled in the same way as Signe on the Where. Maybe 'How to structure your code' or something like that. In addition to the solution provided by Signe.